File: Problems/Leetcode/3001-4000/3201-3300/3265.py
from typing import List


class Solution:
    def countPairs(self, nums: List[int]) -> int:
        n = len(nums)

        def check(i: int, j: int) -> bool:
            if nums[i] == nums[j]:
                return True

            s, t = list(str(nums[i])), list(str(nums[j]))
            if len(s) <= len(t):
                s, t = t, s

            m = len(s)
            for p in range(m):
                for q in range(p + 1, m):
                    s[p], s[q] = s[q], s[p]
                    if int("".join(s)) == int("".join(t)):
                        return True
                    s[p], s[q] = s[q], s[p]
            return False

        cnt = 0
        for i in range(n):
            for j in range(i + 1, n):
                cnt += check(i, j)
        return cnt

File: Problems/Leetcode/3001-4000/3201-3300/3265.cpp
#include <bits/stdc++.h>

using namespace std;
using ll = long long;

class Solution {
public:
    int countPairs(vector<int>& nums) {
        int n = nums.size();

        auto check = [&](int i, int j) {
            if (nums[i] == nums[j]) {
                return true;
            }

            string s = to_string(nums[i]);
            string t = to_string(nums[j]);
            if (s.size() <= t.size()) {
                swap(s, t);
            }

            int m = s.size();
            for (int p = 0; p < m; ++p) {
                for (int q = p + 1; q < m; ++q) {
                    swap(s[p], s[q]);
                    if (stoi(s) == stoi(t)) {
                        return true;
                    }
                    swap(s[p], s[q]);
                }
            }
            return false;
        };

        int cnt = 0;
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                cnt += check(i, j);
            }
        }
        return cnt;
    }
};